File: sdk/javascript/examples/ocr-and-wait.mjs
// IOTYPE_TOKEN=... node examples/ocr-and-wait.mjs contract.pdf
import { Iotype, ProcessingTimeout, fileResult } from "@iotype-ai/sdk";

const path = process.argv[2];
if (!path) { console.error("usage: ocr-and-wait.mjs <file.pdf|file.jpg>"); process.exit(1); }

const io = new Iotype();

const file = await io.ocr(path, { summarize: true });
console.log("uuid:", file.uuid, "(store this — you can resume after a restart)");

try {
  console.log("\n--- extracted text ---");
  console.log(await io.waitFor(file.uuid, { processType: "ocr" }));

  const summary = fileResult(await io.track(file.uuid), "summarize");
  if (summary) console.log("\n--- summary ---\n" + summary);
} catch (e) {
  if (e instanceof ProcessingTimeout) console.log(`still processing; resume with waitFor("${e.uuid}")`);
  else throw e;
}

File: sdk/python/examples/ocr_and_wait.py
"""Upload a PDF for OCR and poll until the text is ready.

    python examples/ocr_and_wait.py contract.pdf
"""
import sys
from iotype import Iotype, ProcessingTimeout

def main(path: str) -> None:
    io = Iotype()

    file = io.ocr(path, summarize=True)
    print(f"uuid: {file.uuid}  (store this — you can resume tracking after a restart)")

    try:
        text = io.wait_for(file.uuid, process_type="ocr", timeout=1800)
    except ProcessingTimeout as exc:
        print(f"still processing; resume later with wait_for({exc.uuid!r})")
        return

    print("\n--- extracted text ---")
    print(text)

    summary = io.track(file.uuid).result("summarize")
    if summary:
        print("\n--- summary ---")
        print(summary)

if __name__ == "__main__":
    if len(sys.argv) < 2:
        raise SystemExit("usage: ocr_and_wait.py <file.pdf|file.jpg>")
    main(sys.argv[1])
